src/dns.cpp

```diff
@@ -342,7 +342,7 @@ uint32_t DNS::compose_name(const uint8_t* ptr, char* out_ptr) const {
             throw dns_decompression_pointer_loops();
         }
         // It's an offset
-        if ((*ptr & 0xc0)) {
+        if (((*ptr & 0xc0) == 0xc0)) {
```

Comment:
Would it make more sense to check for `*ptr == 0xc0`?

Reply:
the lower 6 bits of this byte are the upper bits of the offset, so those bits can be non-zero. the check must for just the upper two bits of the byte

Reply:
Fair enough, I misread what the problem was.